ompi/communicator/comm_init.c

```diff
@@ -169,8 +170,7 @@ int ompi_comm_init(void)
     (void)opal_pointer_array_test_and_set_item(&ompi_mpi_communicators, 0, &ompi_mpi_comm_null);
     (void)opal_pointer_array_test_and_set_item(&ompi_mpi_communicators, 1, &ompi_mpi_comm_null);
 
-    opal_string_copy(ompi_mpi_comm_null.comm.c_name, "MPI_COMM_NULL",
-                     sizeof(ompi_mpi_comm_null.comm.c_name));
+    ompi_mpi_comm_null.comm.c_name = strdup ("MPI_COMM_NULL");
```

Comment:
```suggestion
    ompi_mpi_comm_null.comm.c_name = strdup("MPI_COMM_NULL");
```

Also, I don't know the ordering here: was `ompi_comm_set_nb()` previously invoked, such that this `strdup()` (and the others below) cause a leak?  I.e., wasn't an empty block `malloc()`'ed before this?

Or is this a different code path, and this is the first assignment to `.c_name`?